xray/utils/resultwriter.go

```diff
@@ -3,11 +3,50 @@ package utils
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"github.com/jfrog/jfrog-client-go/utils/errorutils"
 	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
 	"github.com/jfrog/jfrog-client-go/xray/services"
 )
 
+func PrintScanResults(results []services.ScanResponse, isTableFormat, includeVulnerabilities, includeLicenses, isMultipleRoots bool) (err error) {
+	if isTableFormat {
+		var violations []services.Violation
+		var vulnerabilities []services.Vulnerability
+		var licenses []services.License
+		for _, result := range results {
+			violations = append(violations, result.Violations...)
+			vulnerabilities = append(vulnerabilities, result.Vulnerabilities...)
+			licenses = append(licenses, result.Licenses...)
+		}
+
+		if len(results) > 0 {
+			resultsPath, err := WriteJsonResults(results)
+			if err != nil {
+				return err
+			}
+			fmt.Println("The full scan results are available here: " + resultsPath)
+		}
+		if includeVulnerabilities {
+			err = PrintVulnerabilitiesTable(vulnerabilities, isMultipleRoots)
+		} else {
+			err = PrintViolationsTable(violations, isMultipleRoots)
+		}
+		if err != nil {
+			return err
+		}
+		if includeLicenses {
+			err = PrintLicensesTable(licenses, isMultipleRoots)
+		}
+		if err != nil {
+			return err
+		}
+	} else {
+		err = PrintJson(results)
```

Comment:
This seems to be the only usage of the ```PrintJson``` function. Let's therefore move it to this file and make it private.